dali/test/python/test_operator_file_reader.py

```diff
@@ -0,0 +1,60 @@
+import tempfile
+import os
+import nvidia.dali.fn as fn
+from functools import partial
+from nvidia.dali.pipeline import Pipeline
+
+def ref_contents(path):
+    fname = path[path.rfind('/')+1:]
+    return "Contents of " + fname + ".\n";l
+
+def populate(root, files):
+    for fname in files:
+        with open(os.path.join(root, fname), "w") as f:
+            f.write(ref_contents(fname));
+
+def _test_env(func):
+    with tempfile.TemporaryDirectory() as root:
+        files = [str(i)+'.dat' for i in range(10)]
+        populate(root, files)
+        try:
+            func(root, files)
+        finally:
+            for f in files:
+                try:
+                    os.remove(os.path.join(root, f))
+                except:
+                    pass
```

Comment:
I don't think you need try/finally. `with tempfile.TemporaryDirectory() as root:` should remove files AFAIK.